### PR TITLE
Update Thailand holidays: name change for October 13 holiday

### PR DESCRIPTION
**Pull request by PPsyrius**
## Proposed change

This PR adjusted the name for Thailand's October 13 holiday to "วันนวมินทรมหาราช" (HM King Bhumibol Adulyadej Memorial Day) as per [new government announcement on September 26, 2023]( https://www.thaigov.go.th/news/contents/details/72633
), as well as some test case simplification.

## Type of change

- [ ] New country/market holidays support (thank you!)
- [x] Supported country/market holidays update (calendar discrepancy fix, localization)
- [ ] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] I've added references to all holidays information sources used in this PR
- [x] This PR is filed against `beta` branch of the repository
- [x] This PR doesn't contain any merge conflicts and has clean commit history
- [x] The code style looks good: `make pre-commit` command generates no changes
- [x] All tests pass locally: `make test`, `make tox` (we strongly encourage adding tests to your code)
- [x] The related [documentation][docs] has been added/updated (check off the box for free if no updates is required)

[contributing-guidelines]: https://github.com/vacanza/python-holidays/blob/beta/CONTRIBUTING.rst
[docs]: https://github.com/vacanza/python-holidays/tree/beta/docs/source

### CODE REVIEW

File: holidays/locale/be/LC_MESSAGES/BY.po [hunk 1]
```diff
@@ -12,8 +12,7 @@ msgstr ""
 "MIME-Version: 1.0\n"
 "Content-Type: text/plain; charset=UTF-8\n"
 "Content-Transfer-Encoding: 8bit\n"
-"Plural-Forms: nplurals=3; plural=(n%10==1 && n%100!=11 ? 0 : n%10>=2 && n%10<=4 && (n%100<12 || "
```
Comment (KJhellico) [inline holidays/locale/be/LC_MESSAGES/BY.po:15]: It's redundant for this PR (and doesn't match max line length = 99)

Comment (PPsyrius) [reply holidays/locale/be/LC_MESSAGES/BY.po:15]: Oof, that came from `make l10n` and must have been accidentally included in the commit by accident, removing it right now

**Review by KJhellico — APPROVED** [review]
LGTM.

File: holidays/countries/thailand.py [hunk 2]
```diff
@@ -474,14 +474,19 @@ def _populate_public_holidays(self):
...
+                else (
+                    # Anniversary for the Death of King Bhumibol Adulyadej the Great.
+                    tr(
+                        "วันคล้ายวันสวรรคตพระบาทสมเด็จพระบรมชนกาธิเบศร "
+                        "มหาภูมิพลอดุลยเดชมหาราช บรมนาถบพิตร"
+                    )
+                    if self._year >= 2019
+                    # Anniversary for the Death of King Bhumibol Adulyadej.
+                    else tr("วันคล้ายวันสวรรคตพระบาทสมเด็จพระปรมินทรมหาภูมิพลอดุลยเดช บรมนาถบพิตร")
                 )
```
Comment (arkid15r) [inline holidays/countries/thailand.py:489]: I feel this nesting is a bit too much. I suggest using this syntax for simple single level if/else cases:

```
if self._year >= 2023:
  # HM King Bhumibol Adulyadej Memorial Day.
  k_bhumibol_memorial = tr("วันนวมินทรมหาราช")
elif self._year >= 2019:
    # Anniversary for the Death of King Bhumibol Adulyadej the Great.
    k_bhumibol_memorial = tr(
        "วันคล้ายวันสวรรคตพระบาทสมเด็จพระบรมชนกาธิเบศร "
        "มหาภูมิพลอดุลยเดชมหาราช บรมนาถบพิตร"
    )
else:
  # Anniversary for the Death of King Bhumibol Adulyadej.
  k_bhumibol_memorial = tr(
      "วันคล้ายวันสวรรคตพระบาทสมเด็จพระปรมินทรมหาภูมิพลอดุลยเดช บรมนาถบพิตร"
  )
```

Comment (PPsyrius) [reply holidays/countries/thailand.py:489]: Fixed.

On minor note I'm pretty sure Netherland and its oversee territories are also got this format, so we should take a look at them later on.

**Review by arkid15r — APPROVED** [review]
🇹🇭 LGTM

**Pull request merged**